examples/tls-client.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: Copyright (c) 2017-2023 slowtec GmbH <post@slowtec.de>
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Asynchronous TLS client example
```

Comment:
Please add a paragraph with a remark including the link to the original sources you copied the code from here.

Reply:
Done!